### PR TITLE
Skip check for the past dbversion's migration existing if it is squashed

### PR DESCRIPTION
**Pull request by zrhoffman**
This PR
- Fixes an issue introduced by #6096 where, where when migrating from Goose, `db/admin dbversion` would return `0` instead of Goose migration version (e080781e61)
- Fixes the issue that #6096 was supposed to fix: Running the first unsquashed migration without checking whether the past migration exists if migrating from the last squashed migration (f102c654ff).
- Fixes a bug in the Traffic Ops DB tests where `dropdb` did not specify the database to drop

<hr>

## Which Traffic Control components are affected by this PR?

- Traffic Ops

## What is the best way to verify this PR?

1. Delete the `2021080408053529_fix_indices` migrations (unnecessary once #6127 is fixed)
2. Migrate the Traffic Ops database at [ATC 5.1.2](https://github.com/apache/trafficcontrol/commits/RELEASE-5.1.2)
3. Dump the database, move the dump to `traffic_ops_db/test/docker/initdb.d`, and rebuild the `db` service
4. Build a Traffic Ops RPM, move it to `traffic_ops_db/test/docker/traffic_ops.rpm` and rebuild the `trafficops-db-admin` service
5. Run the Traffic Ops DB tests

## If this is a bugfix, which Traffic Control versions contained the bug?

- 6.0.x release branch

## PR submission checklist
- [x] This PR modifies existing tests
- [ ] This PR has documentation 
- [ ] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Comment by zrhoffman** [comment]
Rebased to get the fix for #6127. Removing the `2021080408053529_fix_indices` migration is no longer necessary to test this PR.

File: traffic_ops_db/test/docker/Dockerfile-db-admin [hunk 1]
```diff
@@ -23,25 +23,43 @@ FROM centos:7
...
 
 # Override TRAFFIC_OPS_RPM arg to use a different one using --build-arg TRAFFIC_OPS_RPM=...  Can be local file or http://...
 ARG TRAFFIC_OPS_RPM=traffic_ops.rpm
 ADD $TRAFFIC_OPS_RPM /
-RUN yum install -y \
-        /$(basename $TRAFFIC_OPS_RPM) && \
-        rm /$(basename $TRAFFIC_OPS_RPM) && \
-    yum clean all
+RUN rpm -Uvh $(basename $TRAFFIC_OPS_RPM) && \
+    rm $(basename $TRAFFIC_OPS_RPM)
```
Comment (rawlinp) [inline traffic_ops_db/test/docker/Dockerfile-db-admin:58]: Just for my own edification -- why was this change necessary?

**Comment by rawlinp** [comment]
Uh oh, looks like the DB tests will fail perpetually until they're fixed to ignore the migration that was edited.

**Comment by zrhoffman** [comment]
> Uh oh, looks like the DB tests will fail perpetually until they're fixed to ignore the migration that was edited.

cf59b77ede: No longer failing if that migration is not changed in the PR

File: traffic_ops_db/test/docker/Dockerfile-db-admin (hunk 1 above)
Comment (zrhoffman) [reply traffic_ops_db/test/docker/Dockerfile-db-admin:58]: Installing the TO RPM dependencies in a layer before installing the TO RPM is just an optimization to allow a user to rebuild the `trafficops-db-admin` image with different TO RPMs without needing to download all the Yum packages every time.

Comment (rawlinp) [reply traffic_ops_db/test/docker/Dockerfile-db-admin:58]: Oh ok so `rpm` will still install the required dependencies even if they change in the RPM but aren't updated in the packages above?

Comment (zrhoffman) [reply traffic_ops_db/test/docker/Dockerfile-db-admin:58]: It will not, but if those dependencies are not satisfied when `rpm` runs, `rpm` will fail.

Comment (rawlinp) [reply traffic_ops_db/test/docker/Dockerfile-db-admin:58]: Ok, good to know. So it will be obvious when the TO RPM dependencies change that this list will need to be updated as well. Thanks!

**Review by rawlinp — APPROVED** [review]
